packages/react-ui-core/src/Menu/Menu.js

```diff
@@ -98,11 +93,18 @@ export default class Menu extends PureComponent {
   }
 
   get highlightedOption() {
-    return this.options[this.state.highlightIndex || 0]
+    return this.options[this.state.highlightIndex]
   }
 
-  @autobind
-  handleSelection() {
+  get indexedOptions() {
+    return this.options.map((option, index) => ({
```

Comment:
Would it not make more sense to use `reduce()` here instead of `.map().filter()`?

Reply:
I can change that. I moved this logic from elsewhere. I didn't write it so I didn't look that closely.